### ISSUE TITLE
Output raw text before <!doctype>

### ISSUE THREAD

**Issue by OldStarchy** [mjmlio/mjml#2146]
I'm using this to render templates that are then handled by twig renderer in the cms. In order for the templates to be recognized properly by the cms they need to have some metadata at the start of the file, before the doctype.

```twig
{# subject: New submission from {{ form.name }} #}
{# fromEmail: {{ craft.app.systemSettings.getSettings('email').fromEmail }} #}
{# fromName: {{ craft.app.systemSettings.getSettings('email').fromName }} #}
{# cc: #}
{# bcc: #}
{# includeAttachments: false #}
{# presetAssets: #}
{# description: New form submission - internal notification #}

<!doctype html><html ...
```

I've tried using the `<mj-raw>` tags but the best I can do is at the bottom of the `<head>` tag (as documented).

As it is at the moment I would have to implement some other tool to copy in this metadata from an additional file that would have to be kept in sync per mjml template.

**Comment by iRyusa** [comment]
Not the first time someone need to add something before the HTML document @kmcb777 can we try to add an attribute on mj-raw  in mj-head to add it before the skeleton ?

**Comment by kmcb777** [comment]
Yes good idea, if we use mj-raw for this instead of trying to handle multiple root elements, it should be totally doable

**Comment by kmcb777** [comment]
@iRyusa after further examination, using mj-raw for this would need to tweak mj-raw a little bit, right now we just output the content of all mj-raws inside the mj-head at the same place
so it may be more appropriate to use a new component just for this, more like mj-preview, what do you think ?

**Comment by iRyusa** [comment]
We might need to discuss on the name then as it might be a bit confusing to use 🤔 Or maybe mj-raw inside mjml like mj-head/mj-body?

**Comment by kmcb777** [comment]
yes we could use those, currently those placed here wouldn't be processed at all
maybe with an attribute `position="file-start"` in case there is another use later ?

**Comment by andreas-becker** [comment]
How about an `mj-root` tag allowing to be placed anywhere under `mjml`?
```
<mjml>
  <mj-root before="1"><!-- content before doctype on top of the file --></mj-root>
  <mj-root><!-- content before head tag but inside doctype/html tag --></mj-root>
  <mj-head>...</mj-head>
  <mj-root><!-- content between head and body tag --></mj-root>
  <mj-body>...</mj-body>
  <mj-root><!-- content after body tag --></mj-root>
  <mj-root after="1"><!-- content after closing html tag--></mj-root>
</mjml>
```

or as another suggestion, `before/after` could be replaced with `outside`

**Comment by JamesWilson3** [comment]
Hello, I checked the documentation https://documentation.mjml.io/#mj-raw and there doesn't seem to be anything for outputting raw text before <!doctype> yet. Just wondering if there is some sort of workaround available for now?

**Comment by iRyusa** [comment]
👍 We can close this as it's now available in 4.12
